vcr/vcr.go

```diff
@@ -297,26 +293,61 @@ func (c *vcr) Issue(template vc.VerifiableCredential) (*vc.VerifiableCredential,
 	return &credential, nil
 }
 
-func (c *vcr) Resolve(ID ssi.URI) (*vc.VerifiableCredential, error) {
+func (c *vcr) Resolve(ID ssi.URI, resolveTime *time.Time) (*vc.VerifiableCredential, error) {
 	credential, err := c.find(ID)
 	if err != nil {
 		return nil, err
 	}
 
-	revoked, err := c.isRevoked(ID)
+	if err = c.validateForUse(credential, resolveTime); err != nil {
+		switch err {
+		case ErrRevoked:
+			return &credential, ErrRevoked
+		case ErrUntrusted:
+			return &credential, ErrUntrusted
+		default:
+			return nil, err
+		}
+	}
+	return &credential, nil
+}
+
+func (c *vcr) validateForUse(credential vc.VerifiableCredential, validAt *time.Time) error {
+	revoked, err := c.isRevoked(*credential.ID)
 	if revoked {
-		return &credential, ErrRevoked
+		return ErrRevoked
 	}
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	trusted := c.isTrusted(credential)
 	if !trusted {
-		return &credential, ErrUntrusted
+		return ErrUntrusted
 	}
 
-	return &credential, nil
+	return c.validate(credential, validAt)
+}
+
+func (c *vcr) validate(credential vc.VerifiableCredential, validAt *time.Time) error {
+	at := time.Now()
+	if validAt != nil {
+		at = *validAt
+	}
+	issuer, err := did.ParseDIDURL(credential.Issuer.String())
+	if err != nil {
+		return err
+	}
+
+	if credential.IssuanceDate.After(at) {
```

Comment:
Nitpick: should this be after or equal or only after?

Reply:
Given that `at` would mostly be `now` and `IssuanceDate` only has an accuracy on the seconds, equal will do nothing.